docs/composite-clusters.md

```diff
@@ -0,0 +1,129 @@
+
+##### Scaling
+
+In this scenario, the ipfs-cluster setup has reached a number of peers which degrades the performance of the cluster. By spliting a single cluster into several composite subclusters, it allows to reduce the number of peers participating in the consensus and thus, scaling the cluster.
+
+
+
+### Each cluster peer attached to a peer from other but unique cluster (cylinder clusters)
+
+In this topology, each cluster peers uses as ipfs daemon another cluster peer from a different cluster, but all underlying peers belong to the same cluster.
+
+Todo: drawing
+
+#### Uses
+
+##### Multimetric allocation
+
+In this case, each cluster uses a different allocator/informer metric. The overcluster would allocate content to the undercluster based on the first metric. The undercluster may use the information from the first allocation as input to another allocator with different metric. In order for this use-case to be useful, such special allocator would need to exist.
+
+##### Double-cluster
+
+In this case (and given an allocate-everywhere replication factor), the obercluster and the undercluster would maintain the same state. The overcluster could be take offline, be upgraded, point to the real ipfs daemons and replace the undercluster, which could then be taken offline. Of course, pinning and unpinning operations would not need to be allowed when there is no overcluster, or manual sync of the state should be needed when it comes back up. It is not very clear if this is a useful usecase.
+
+##### Pyramid pinning
+
+In a cylinder cluster, pin requests to the overcluster are tracked by all the underclusters below. Pin requests to the underclusters, are not tracked by the overcluster, but everything is eventually pinned in the ipfs daemons at the bottom. This allows to build pinning hierarchies, where the bottom clusters have visibility to larger pinsets than the top clusters.
+
+This is a specific case of **inverse multi-ring topology** described below, with a single overcluster.
+
+### Multiple clusters have their cluster peers attached to peers from a single cluster (inverse multi-ring)
+
+In this topology, we have several overclusters which all share the same undercluster.
+
+#### Uses
+
+##### Data partitioning
+
+A group of parties may which to share a common cluster. With an inverse multi-ring setup, each party will mantain their own pinset in their own cluster, but it will use a common infrastructure (the undercluster), which is shared with others. Each overcluster may administer how cluster APIs are exposed and what authentication to use.
```

Comment:
I am having a hard time understanding why this is something people would want to do.  In what situation would this be desirable over everyone banding together into a single cluster?

Reply:
Basically, maintaining a separate pinset, not worrying about ipfs setup, and being able to take down a cluster without affecting other people/clusters. Of course, most of it is just delegating problems to the person maintaining the under cluster.